cli/azd/pkg/project/container_helper.go

```diff
@@ -216,6 +221,52 @@ func (ch *ContainerHelper) Deploy(
 ) *async.TaskWithProgress[*ServiceDeployResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServiceDeployResult, ServiceProgress]) {
+			if dotnetDetails, ok := packageOutput.Details.(*dotNetSdkPublishConfiguration); ok {
+				task.SetProgress(NewServiceProgress("Logging into container registry"))
+				dockerCreds, err := ch.Credentials(ctx, serviceConfig, targetResource)
+				if err != nil {
+					task.SetError(fmt.Errorf("getting docker credentials: %w", err))
+					return
+				}
+
+				task.SetProgress(NewServiceProgress("Building and pushing container image"))
+				imageName := fmt.Sprintf("azd-deploy-%s-%d", serviceConfig.Name, time.Now().Unix())
```

Comment:
ContainerHelper already contains functions that can be reused for proper image name generation based on all our current configurable rules.